feat(kinput): component reskinning [KHCP-8985] (#1759)
Kong/kongponents: docs/guide/migrating-to-version-9.md

### KInput

#### Structure

* `form-control` class is no longer present
* `over-char-limit` class is no longer present

> * `has-error` class is no longer present

Reviewer: question: did this one get replaced? If not, please change as shown here:
```suggestion
* `has-error` class has been removed
```

* `help` class has been changed to `help-text`

#### Constants, Types & Interfaces

* `SizeArray` and `IconPositionArray` constants are no longer exported

> * `Size` and `IconPosition` types are no longer exported

Reviewer: question: are they no longer exported, or have been removed?

Author: Changed this to `have been removed`

#### Props

* `overlayLabel` prop is no longer supported
* `size` prop is no longer supported

> * `iconPosition` prop is no longer supported

Reviewer: "has been removed" instead of supported - I'd also state next to the `size` prop that all inputs are a single size

#### Slots

> * `icon` slot is no longer supported. Instead, you can use new `before` and `after` slots

Reviewer: ```suggestion
* `icon` slot has been removed. Instead, you can use new `before` and `after` slots
```

#### Events

> No changes.

Reviewer: nit: if no changes, this section doesn't need to exist

Reviewer: The same order is fine, but if a section isn't needed, there's no reason to clutter up the docs with empty header sections.
